crontab/notify_old_pp.php

```diff
@@ -5,98 +5,107 @@
+        $time_details = sprintf(_('[checked out since %1$s, project last visited %2$s]'), $modifieddate, $lastvisitdate);;
+        $projects_list[] = "$work_details\n$time_details\n    $code_url/project.php?id=$projectid";
     }
 
-    $rownum++;
+    $projects_list_string = implode("\n\n", $projects_list);
+
+    $numprojs = count($projects);
+    echo "\nNotifying $PPer about $numprojs project(s):\n$projects_list_string\n\n";
+
+    $message = [];
+    if($which_email == "first")
+    {
+        // TRANSLATORS: %s is the site abbreviation (eg: 'DP')
+        $subject = sprintf(_("%s: Renew or Return Post-Processing Projects"), $site_abbreviation);
+
+        $message[] = sprintf(_("Hello %s,"), $PPer);
+        $message[] = _("This is an automated message.");
+        $message[] = sprintf(_("Our database system indicates that you have had one or more projects checked out for Post-Processing for more than %d days:"), $pp_alert_threshold_days);
+        $message[] = $projects_list_string;
+        $message[] = _("It is important that our books move relatively quickly through the Post-Processing process so that they may be posted at Project Gutenberg. We have a lot vested in each project -- each represents many hours of volunteer work. We are all eager to see each book posted and there is also a risk that that work would be lost should other organizations produce and post a book before we finish.");
+        $message[] = '==' . _("Do you wish to continue or to stop working on these books?") . "==";
+        $message[] = _("If you haven't yet finished and wish to continue working on these books, please visit the project pages listed above. This will update the status of the projects.");
+        $message[] = _("If you no longer wish to have these projects assigned to you, please visit the project pages and select 'Return to Available', or reply to this email and state that you no longer wish to have the projects in question assigned to you so that we may return them to the pool for someone else to work on.");
+        $message[] = "==" . _("Have you completed your work?") . "==";
+        $message[] = _("If you have completed your post-processing work, please visit the appropriate link listed above, select the 'Upload for verification' option on the Project Page, and follow the prompts. If you have any special information or comments to pass on, you will be able to leave a message for the verifier during this process. If you have Direct Upload capability and no longer use Post-Processing Verification, please upload each completed project directly to PG.");
+        $message[] = "==" . _("Are there any problems?") . "==";
+        $message[] = sprintf(_("If any project is missing page images or illustrations, please contact the Project Manager (PM) for assistance. If the PM does not respond within a reasonable amount of time, please contact db-req at %s concerning any problems."), $db_requests_email_addr);
```

Comment:
The name "db-req" is pgdp-specific, isn't it? It looks like we have hard-coded it in a few places, but not many. Oh well, PPing is pgdp-specific in the first place.

Reply:
Yes, this is more pgdp.net creep into the codebase. I've (almost) stopped fighting these little intrusions. <sigh>

Reply:
I have no objection to changing the wording from `please contact db-req at %s concerning` to `please email %s and request help concerning`. I need to occasionally be reminded to try to agnosticize (is that a word) what goes into the code.

Reply:
Nicely done - updated.

Reply:
Yeah, I was going to suggest something like that, but noticed there were also occurrences of "db-req" in the mid-month message, so decided not to bother.

Reply:
That wording can be tweaked, too. How about:
> $message[] = sprintf(_("If you have already notified %s of an expected absence, you may safely ignore this message. Otherwise, any projects in your queue for which a previous notice has been sent may be reclaimed if you do not respond or renew within the next 7 days."), $db_requests_email_addr);

Post-processing *can* be a pgdp-specific thing, but needn't be. If it is, then whoever is using the code won't need this cron, anyway. If it isn't, they may need something like this.

Reply:
Looks good with the new wording, thanks!